front/src/modules/simulationResult/hooks/useFormattedOperationalPoints.ts

```diff
@@ -16,28 +17,28 @@ import { formatOperationalPoints } from '../SimulationResultExport/utils';
 export const useFormattedOperationalPoints = (
   train?: TrainScheduleBase,
   simulatedTrain?: SimulationResponseSuccess,
-  pathProperties?: PathPropertiesFormatted,
-  infraId?: number
+  pathProperties?: PathPropertiesFormatted
 ) => {
   const [operationalPoints, setOperationalPoints] = useState<OperationalPointWithTimeAndSpeed[]>();
   const [loading, setLoading] = useState(false);
+  const { getTrackSectionsByIds } = useScenarioContext();
 
   useEffect(() => {
-    if (train && simulatedTrain && pathProperties && infraId) {
+    if (train && simulatedTrain && pathProperties) {
       const fetchOperationalPoints = async () => {
         setLoading(true);
         const formattedOperationalPoints = await formatOperationalPoints(
           pathProperties.operationalPoints,
           simulatedTrain,
           train,
-          infraId
+          getTrackSectionsByIds
```

Comment:
Would it be possible to get the `trackSections` outside the function `formatOperationalPoints` and to only give the trackSections as argument ?

Reply:
I think so!